sw/device/sca/aes_serial.c

```diff
@@ -66,6 +93,34 @@ static void aes_serial_set_key(const uint8_t *key, size_t key_len) {
   SS_CHECK_DIF_OK(dif_aes_start(&aes, &transaction, &key_shares, NULL));
 }
 
+/**
+ * Set key with shares
+ *
+ * This function uses key shares.
+ * The key must be `kAesKeySize` bytes long.
+ *
+ * @param key Key.
+ * @param key_len Key length.
+ * @return Result of the operation.
+ */
+static void aes_serial_set_key_share(const uint8_t *key, size_t key_len) {
+  SS_CHECK(key_len == kAesKeyLength);
+  dif_aes_transaction_t transaction = {
+      .operation = kDifAesOperationEncrypt,
+      .mode = kDifAesModeEcb,
+      .key_len = kDifAesKey128,
+      .masking = kDifAesMaskingInternalPrng,
+      .manual_operation = kDifAesManualOperationManual,
+  };
```

Comment:
Can you set it up just once when starting the capture and then pass a pointer to this to the function?